### ISSUE TITLE
why zero_grad the loss model after the evaluation step ?

### ISSUE THREAD

**Issue by Mogady** [UKPLab/sentence-transformers#1530]
Hi, I'm trying to implement loss evaluator and integrate  tesnorboard, while navigating your code I noticed you zero_grad the loss model after each evaluation step. I don't get it, since optimizer.zero_grad have the same effect right? so why only zero grading the loss model after the evaluation step and not the training step?

https://github.com/UKPLab/sentence-transformers/blob/40af04ed70e16408f466faaa5243bee6f476b96e/sentence_transformers/SentenceTransformer.py#L728

**Comment by nreimers** [comment]
It is there just as a caution if an evaluator computes a gradient by accident.

After each optimizer step the gradient is resettled too
